### PR TITLE
Restructure the repository by category (PR #1)

### LINKED ISSUE #688: Reorganize directories to match GKE top-level topics
Historically, each sample lived in their own top-level directory with few exceptions.

With the growing amount of samples, this is quickly becoming unsustainable. I propose we move to storing samples in top-level topic names instead.

For example:
- All AIML samples are stored under `ai-ml/`
- Similarly for `batch/`, `autopilot`, `security`, etc.

This would also require moving existing samples to appropriate directories, which means modifying those samples' documentation (in c.g.c) to point to new directories.

### PR DESCRIPTION
**Pull request by bourgeoisor**
This PR restructures the GKE samples by making sure that each sample lives in a subdirectory relating to its main point of interest (security, observability, networking, etc.)

To achieve this, we need to operate in 3 distinct steps:
- **Step 1**: duplicate each sample into its proper category sub-directory, and make sure that each triggers and workflows point to these new locations (this PR)
- **Step 2**: modify each GKE core tutorials utilizing these samples so they point the users to their new locations
- **Step 3**: finally, delete the old locations / duplicates of each samples

Fixes #688 

## What will this repo look like post-restructure?
Once each steps of the restructure are over, the root of this repo will look like this:
```
.github/
ai-ml/
autopilot/
batch/
cost-optimization/
databases/
demos/
management/
networking/
observability/
quickstarts/
security/
streaming/
windows/
.gitignore
LICENSE
README.md
```

## Which samples moved?
Some samples were already contained in category sub-directories. Those samples will not have to move.

This is the full list of moves contained in this PR:
```
cloud-pubsub -> databases
cloudsql -> (removed, this was already empty)
custom-metrics-autoscaling -> observability
gke-scheduled-autoscaler -> cost-optimization
gke-stateful-kafka -> streaming
gke-stateful-mysql -> databases
gke-stateful-postgres -> databases
gke-vpa-recommendations -> cost-optimization
guestbook -> quickstarts
hello-app -> quickstarts
hello-app-cdn  -> quickstarts
hello-app-redis -> quickstarts
hello-app-tls -> quickstarts
load-balancing -> networking
migrating-node-pool -> management
monitoring/distributed-tracing -> observability
network/dpv2-observability -> networking
network-policies -> networking
quickstart/<lang> -> quickstarts/languages
stateful-workload-filestore -> databases
terraform -> .github (these aren't GKE samples so I'm hiding it away)
try-gke -> demos
whereami -> quickstarts
windows-multi-arch ->  windows
wordpress-persistent-disks ->  quickstarts
workload-metrics -> observability
```

## How to review?
Considering this PR modifies hundreds of files and tens of thousands of lines, I recommend reviewing by reviewing one commit at a time, skipping the commit entitled "Duplicate samples in new locations" (this is the one that creates duplicates of the samples into their "new" locations). Alternatively, you can review the 3 important non-sample-duplicate commits by [clicking here](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/987/files/16f1a0df1bd19433502571e88af17fbfb5d09063..6580db6c5530c3f8b452c13ee028eb59f50c9177).

**Note**: Ignore the license check failures. The check is assuming that all of these added files are new, but they're old samples (duplicated).

### CODE REVIEW

**Comment by NimJay** [comment]
Summary: let's **ignore `header-check` for this pull-request**.
Sorry, I didn't do a full/proper review of this pull-request yet.
But just a heads-up for everyone: let's ignore the `header-check` when we do eventually merge this restructuring pull-request.
Adding licenses headers to files missing license headers is out-of-scope for this pull-request.

Thank you, Olivier, for initiating this work! I'm exciting for this massive clean-up. 💯

File: windows-multi-arch/README.md
Comment (NimJay) [inline windows-multi-arch/README.md:3]: **Praise:** Cool use of `[!IMPORTANT]`!

File: .github/CONTRIBUTING.md
Comment (NimJay) [inline .github/CONTRIBUTING.md:1]: **Question:** Will we update the [`CODEOWNERS` file](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/.github/CODEOWNERS) in a separate pull-request? (I think a separate pull-request makes sense, since this change is already quite large.)

File: .github/terraform/google-cloud-build-triggers.tf
Comment (NimJay) [inline .github/terraform/google-cloud-build-triggers.tf:1]: **Question:** Do you plan on re-applying this Terraform immediately after merging this pull-request?

File: .github/workflows/quickstarts-guestbook-ci.yml
Comment (NimJay) [inline .github/workflows/quickstarts-guestbook-ci.yml:1]: **Praise:** I like that we're just prefixing the `.github/workflows` files with `quickstarts`/`observability`/etc. instead of introducing more sub-directories (like `.github/workflows/quickstarts`) — keeping the repo shallow?

File: quickstarts/wordpress-persistent-disks/README.md
Comment (NimJay) [inline quickstarts/wordpress-persistent-disks/README.md:1]: **Thought:** (no action needed)
`quickstarts/` feels like an awkward place for `wordpress-persistent-disks`.
But it's best folder choice.
`databases/` wouldn't make sense for it either.

**Review by NimJay — APPROVED** [review]
Looks good to me! Thanks again for driving this effort. 👏 
Feel free to merge.

File: quickstarts/wordpress-persistent-disks/README.md
Comment (bourgeoisor) [reply quickstarts/wordpress-persistent-disks/README.md:1]: I was torn with that one, mostly because that sample is for one of the 2 "quickstart tutorials" in our GKE docs, so I decided with "quickstarts/" at the end of the day

File: .github/workflows/quickstarts-guestbook-ci.yml
Comment (bourgeoisor) [reply .github/workflows/quickstarts-guestbook-ci.yml:1]: Was mostly cleaning up. If it becomes too unwieldy we can always do subdirectories there too, but on the bright side doing _that_ change wouldn't affect anything else (no code changes, no docs changes).

File: .github/terraform/google-cloud-build-triggers.tf
Comment (bourgeoisor) [reply .github/terraform/google-cloud-build-triggers.tf:1]: Yes.

File: .github/CONTRIBUTING.md
Comment (bourgeoisor) [reply .github/CONTRIBUTING.md:1]: Oh good catch on CODEOWNERS, that's an unintentional omission. We can do separate PR, I'll note it down.

File: .github/terraform/google-cloud-build-triggers.tf
Comment (NimJay) [reply .github/terraform/google-cloud-build-triggers.tf:1]: Just noticed the checklist in [your one-pager](http://go/dee-po-1pager-gke-samples-restructuring). That helps!

**Pull request merged**
